aicssegmentation/util/lazy.py

```diff
@@ -0,0 +1,16 @@
+# Lazy evaluation utils
+
+
+def lazy_property(fn):
```

Comment:
just curious about what does "lazy_property" mean?

Reply:
Lazy refers to [Lazy initialization](https://en.wikipedia.org/wiki/Lazy_initialization). This is just a handy decorator to turn a property automatically into a lazy initialized property. In concrete terms it means that the code within the decorated property will 1) only be run once when the property is accessed for the first time and 2) the result is stored in a variable and will be used as return value for subsequent access. In other words, the code is only used to "create" the object.

This is really just a simple trick to avoid repeating boilerplate code like this:
~~~python
class WorkflowDefinition:

    self._thumbnail_pre = None

    def thumbnail_pre(self) -> np.ndarray:
        if self._thumbnail_pre is None:
              self._thumbnail_pre =  np.squeeze(imread(Directories.get_assets_dir() / f"thumbnails/{self.name.lower()}_pre.png"))
        return self._thumbnail_pre
~~~
And turn it into this:
~~~python
class WorkflowDefinition:

    @lazy_property
    def thumbnail_pre(self) -> np.ndarray:        
        return np.squeeze(imread(Directories.get_assets_dir() / f"thumbnails/{self.name.lower()}_pre.png"))
~~~